### DISCUSSION TITLE
How to notify the service user to play a sound / enfore certain interpret config options

### DISCUSSION THREAD

**Discussion by pke** [statelyai/xstate#2661] [Q&A]
Take the classic alarm clock with its "beep" activity. As the user of this state chart I'd like to provide the machine with the function to actually play a beep sound. Depending on the environment the service runs in, this would happen in different ways, that the machine should not be bothered with. Instead I'd like this side effect to be handled outside the machine.

I guess the correct way is to provide either an action or activity as init config for the machine interpretation? If I am on the right track then how can the machine typescript configuration enforce the user of the machine to supply such function?

**Comment by mattpocock** [comment]
```ts
const makeMachine = (beep: () => void) => createMachine({ entry: beep });
```

**Reply by pke** [reply]
This machine will call beep when its created?

Also the typescript aspect was not answered. I want to ensure the machine does not call a non existing entry/activity. But I guess when working with strings in the machine definition, it will just print a warning when the entry action or activitiy was not supplied during machine creation?

**Reply by davidkpiano** [reply]
Currently, there will only be a warning when the action is not supplied (activities are deprecated). In the future, there will be type errors if implementations are missing.
